src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -235,21 +284,238 @@ export class Model3DViewer extends React.Component<Model3DViewerProps> {
     }
   };
 
+  slice = (sliceProps: SlicingProps) => {
+    if (this.viewer) {
+      const planes = this.state.planes;
+      if (sliceProps.x) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(sliceProps.x.direction ? 1 : -1, 0, 0),
+          sliceProps.x.coord
+        );
+        planes[0] = plane;
+      }
+      if (sliceProps.y) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, sliceProps.y.direction ? 1 : -1, 0),
+          sliceProps.y.coord
+        );
+        planes[1] = plane;
+      }
+      if (sliceProps.z) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, 0, sliceProps.z.direction ? 1 : -1),
+          sliceProps.z.coord
+        );
+        planes[2] = plane;
+      }
+      this.setState({ planes });
+    }
+  };
+
+  takeScreenShot = async () => {
+    if (this.viewer) {
+      const url = await this.viewer.getScreenshot();
+      if (this.props.onScreenshot) {
+        this.props.onScreenshot(url);
+      }
+    }
+  };
+
+  onchange = (val: number, axis: string) => {
+    const planes = this.state.planes;
+    const flipped = this.state.flipped;
+    switch (axis) {
```

Comment:
Here you could also do something like:
```js
const axisToIndexMap = {x: 0, y: 1, z: 2};
const index = axisToIndexMap[axis];
planes[index].set(planes[index].normal, flipped[index] ? val : -val);
```
and then remove the entire switch-statement

Reply:
Removed